pkg/snapstore/s3_snapstore.go

```diff
@@ -495,30 +495,40 @@ func (s *S3SnapStore) Delete(snap brtypes.Snapshot) error {
 	return err
 }
 
-// S3SnapStoreHash calculates and returns the hash of aws S3 snapstore secret.
-func S3SnapStoreHash(config *brtypes.SnapstoreConfig) (string, error) {
-	if dir, isSet := os.LookupEnv(awsCredentialFile); isSet {
-		awsConfig, err := readAWSCredentialFromDir(dir)
+// GetS3CredentialsLastModifiedTime returns the latest modification timestamp of the AWS credential file(s)
+func GetS3CredentialsLastModifiedTime() (time.Time, error) {
+	if dir, isSet := os.LookupEnv(awsCredentialDirectory); isSet {
+		awsTimeStamp, err := getAWSBackupCredentialLatestModifiedTime(dir)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials from %v directory", dir)
+			return time.Time{}, fmt.Errorf("failed to get AWS credential timestamp from the directory %v with error: %v", dir, err)
 		}
-		return getS3Hash(awsConfig), nil
+		return awsTimeStamp, nil
 	}
 
 	if filename, isSet := os.LookupEnv(awsCredentialJSONFile); isSet {
-		awsConfig, err := credentialsFromJSON(filename)
+		fileInfo, err := os.Stat(filename)
 		if err != nil {
-			return "", fmt.Errorf("error getting credentials using %v file", filename)
+			return time.Time{}, fmt.Errorf("failed to fetch file information of the AWS JSON credential file %v with error: %v", filename, err)
 		}
-		return getS3Hash(awsConfig), nil
+		return fileInfo.ModTime(), nil
 	}
 
-	return "", nil
+	return time.Time{}, fmt.Errorf("no environment variable set for the AWS credential file")
 }
 
-func getS3Hash(config *awsCredentials) string {
-	data := fmt.Sprintf("%s%s%s", config.AccessKeyID, config.SecretAccessKey, config.Region)
-	return getHash(data)
+// getAWSBackupCredentialLatestModifiedTime reads the modification timestamp of each AWS
+// credential file in the credential directory and returns the latest
+func getAWSBackupCredentialLatestModifiedTime(dirname string) (time.Time, error) {
+	_, err := os.Stat(dirname)
+	// the access credential directory has errored
+	if err != nil {
+		return time.Time{}, err
+	}
```

Comment:
this part of code is keep getting repeated in each function, why not put this inside `func getLatestFileModificationTime()` as well.

Reply:
Addressed by commit [51ae353](https://github.com/gardener/etcd-backup-restore/pull/670/commits/51ae3533ad16c589fa8bf859ec52cb0da7b392c5)